### PR TITLE
[WIP]Open in app - Universal Links

### PR DESCRIPTION
**Pull request by pablocarmu**
BUGS & IMPROVEMENTS:

- [X] (1) Settings view in foreground https://github.com/owncloud/ios/pull/999#issuecomment-364112495 [FIXED]
- [X] (2) Error message https://github.com/owncloud/ios/pull/999#issuecomment-364115751  [FIXED]
- [X] (3) Recent shared files are not opened https://github.com/owncloud/ios/pull/999#issuecomment-364380066 [FIXED]
- [X] (4) Crash when file is not accesible https://github.com/owncloud/ios/pull/999#issuecomment-364382928 [FIXED]
- [X] (5) Crash in file contained in folder with special characters https://github.com/owncloud/ios/pull/999#issuecomment-364389445  [FIXED]
- [X] (6) Error message in folder with special characters https://github.com/owncloud/ios/pull/999#issuecomment-364389667  [FIXED]
- [X] (7) Folder name with blanks https://github.com/owncloud/ios/pull/999#issuecomment-364413022 [FIXED]
- [X] (8) File downloaded and not opened https://github.com/owncloud/ios/pull/999#issuecomment-364447851 [FIXED]
- [X] (9) Crash when opening pics https://github.com/owncloud/ios/pull/999#issuecomment-375204366 [NOT REPRODUCIBLE]
- [X] (10) When app is opened after being killed, links not opened [FIXED]

### CODE REVIEW

**Comment by jesmrec** [comment]
### BUG (1)

Steps

1. Get the private link of a non-supported-format file in the app
2. In the app, open Settings view
3. Vanish the app (not kill)
4. Tap on the link

Current: App opened, and settings view is displayed instead of files view. The highlighted file is not visible
Expected: Files view visible, and file highlighted.

Tested with iPadAir v11

CC @pablocarmu

**Comment by jesmrec** [comment]
### BUG (3)

Steps

1. App, login with user1
2. In webUI, user2 shares a file with user1 (get the link)
3. In device, in notes app (for example) click on the link

Current: App opened, with message: "Unable to open this file. Please make sure it exists"
Expected: File downloaded and opened

Tested with iPadAir v11

**Comment by jesmrec** [comment]
### BUG (4)

Steps

1. App, login with user1
2. In webUI, user2 shares a file with user1 (get the link)
3. User2 removes the shared item in web UI
3. In device, user1 in notes app (for example) click on the link

Current: Crash. File is not accesible for this user.
Expected: Error message

Tested with iPadAir v11

**Comment by jesmrec** [comment]
### BUG (5)

Steps

1. Take the private link of a file contained in a folder with special characters (i have used this one: level7$%&éâ)
2. Click on the link in notes app

Current: App crashes. 
Expected: File opened / highlighted

Tested with iPadAir v11

**Comment by jesmrec** [comment]
### BUG (6)

Steps

1. Take the private link of the folder in previous https://github.com/owncloud/ios/pull/999#issuecomment-364389445
2. Click on the link in notes app

Current:  Generic error shown. 
Expected: Folder opened

Tested with iPadAir v11

**Comment by jesmrec** [comment]
### BUG (7)

Steps

1. Create a folder which name contains a blank, for example "Doc uments"
2. Take the private link and click on it

Current:  Generic error shown. 
Expected: Folder opened

Tested with iPadAir v11

**Comment by jesmrec** [comment]
### BUG (8)

Steps

1. Login in the app and download a format-supported file
2. Switch internet connection off
3. Click on the private link

Current:  Generic error shown. 
Expected: File was downloaded locally, and can be opened and displayed 

Tested with iPadAir v11

**Comment by pablocarmu** [comment]
I've been researching about https://github.com/owncloud/ios/pull/999#issuecomment-364389445 and https://github.com/owncloud/ios/pull/999#issuecomment-364389667 and this happened because of the encoding/decoding of the `Location` header.

I've fixed the encoding but the bug still happens when the file/folder has `%` or `&` on its name.
This is going to change when we start using the new header, which i'm working on it now.

@jesmrec

**Comment by jesmrec** [comment]
in which point are we here @pablocarmu ?

**Comment by pablocarmu** [comment]
@jesmrec Right now all the bugs you reported in this PR are ready to test. Let me know if one or more still happens.

**Comment by jesmrec** [comment]
@pablocarmu please rebase, if posible against 3.7.3 tag

**Comment by jesmrec** [comment]
### BUG (9)

Steps

1. Take the private link of a pic in second level of deep, that is downloaded.
2. Switch internet connection off
3. Click on the private link

Current:  Crash
Expected: Pic is opened

Not always reproducible

Tested with iPhone6SPlus v11

File: Owncloud iOs Client/Tabs/FileTab/FilesViewController.m [hunk 1]
```diff
@@ -234,6 +234,23 @@ - (void) viewDidAppear:(BOOL)animated {
             self.navigationItem.titleView = imageView;
         }
     }
+    
+
+//    //If is a new user set the file list
+//    if (app.isNewUser) {
```
Comment (javiergonzper) [inline Owncloud iOs Client/Tabs/FileTab/FilesViewController.m]: This code can not be commented. It is used to put the tableView as it should after a change of user

File: Owncloud iOs Client/Tabs/FileTab/FileListDBOperations.m [hunk 2]
```diff
@@ -164,6 +164,46 @@ + (void) createAFolder: (NSString *)folderName inLocalFolder:(NSString *)localFo
...
 }
 
+/*
+ * This method receive the new array of the server and store the changes
+ * in the Database and in the tableview
+ * @param requestArray -> NSArray of path items
+ */
++(void)deleteOldDataFromDBBeforeRefresh:(NSArray *) requestArray parent:(FileDto *) parentFileDTO {
+
+    UserDto *activeUser = [ManageUsersDB getActiveUser];
```
Comment (javiergonzper) [inline Owncloud iOs Client/Tabs/FileTab/FileListDBOperations.m]: You can get the active user with:
`APP_DELEGATE.activeUser`

File: Owncloud iOs Client/Tabs/FileTab/FileListDBOperations.m [hunk 3]
```diff
@@ -164,6 +164,46 @@ + (void) createAFolder: (NSString *)folderName inLocalFolder:(NSString *)localFo
...
+    NSMutableArray *directoryList = [NSMutableArray arrayWithArray:requestArray];
+
+    NSString *subfolders = [UtilsUrls getServerSubfolders: [ManageUsersDB getActiveUser]];
```
Comment (javiergonzper) [inline Owncloud iOs Client/Tabs/FileTab/FileListDBOperations.m]: Here you are getting the active user again when you got it a few lines behind doing two sql requests

**Pull request merged**